### ISSUE TITLE
[ur_description] base meshes have wrong orientation

### ISSUE THREAD

**Issue by fmessmer** [ros-industrial/universal_robot#503]
given the discussion in https://forum.universal-robots.com/t/dynamically-changing-gravity-vector/7633/6, 

```
[quote="felixmessmer, post:5, topic:7633, full:true"]
@Ebbe
so the wire is aligned with the positve y-axis of the gravity reference coordinate system?
[/quote]
[quote="Ebbe, post:6, topic:7633, full:true"]
Yes, the wire is aligned with y-axis of the robot base coordinate system.
[/quote]
```

it seems the meshes for the base (i.e. `base.dae` and `base.stl`) have wrong orientation...the cables are aligned with the negative y-axis of `base` link - but should be aligned with positive y-axis of `base` link

this affects only the non-e-series robots....e-series robots do not have a cable modeled...

the coordinate systems - mainly `base_link` and `base` - are correct, but I think the meshes should be rotated by 180°

---
can someone help me to verify this?
it's a bit of a brain twister with the various coordsystems in ROS and only semi-optimal display of axis on the TeachPanel

**Comment by gavanderhoorn** [comment]
This is a known issue, and doesn't really affect anything other than the visualisation.

See:

https://github.com/ros-industrial/universal_robot/blob/06d8b9e2f5f86aa54f9f2845f11edbc84e2f951e/ur_description/urdf/ur3.urdf.xacro#L347-L351

and:

https://github.com/ros-industrial/universal_robot/blob/4215b697e0955f3ec78d5c0ce6261348164d8616/ur_description/urdf/ur_macro.xacro#L78-L84

and:

https://github.com/ros-industrial/universal_robot/blob/4215b697e0955f3ec78d5c0ce6261348164d8616/ur_description/urdf/ur_macro.xacro#L321-L325

I'm pretty sure we already had an issue for this, but I can't find it, so I'll keep this open for now.

> it's a bit of a brain twister with the various coordsystems in ROS and only semi-optimal display of axis on the TeachPanel

The only thing you should compare between the ROS model and the TP is the `base -> tool0` transform. Anything else will not be correct.

> given the discussion in https://forum.universal-robots.com/t/dynamically-changing-gravity-vector/7633/6,

please copy the relevant parts here, as we cannot read that discussion without registering for that site. I would not know what you are referring to.

**Comment by fmessmer** [comment]
> This is a known issue, and doesn't really affect anything other than the visualisation.

yes, it's just visualization - the tf frames `base` (ur-internal), `base_link` (ROS-bw-compatibility) as well as the others are correct
but it's misleading, if you get hints from the support forum as posted above and then find the mesh to be oriented in a different way

> NOTE: this WILL be removed in a future revision of this file 

why not now together with all the other "breaking" changes?
or at least make it configurable - via yaml?



however, I don't know whether this is generally wrong or depend on robot type and/or individual settings for each robot...

**Comment by gavanderhoorn** [comment]
> but it's misleading

"misleading" is a bit strong. It's confusing. It's not like this has been purposefully modelled like this, to make things harder for everyone else.

> if you get hints from the support forum as posted above

I still wouldn't know what you are referring to, as we cannot read the discussion.

> > NOTE: this WILL be removed in a future revision of this file
> 
> why not now together with all the other "breaking" changes?

Because I haven't had time for it yet, together with the many other things that still need to be done.

And I want to do it in its own PR, not as part of some other PR.

The comment is just there as a fail-safe: in case this is forgotten, or people start using `melodic-devel-staging` before they should, no one can complain when it *is* fixed at some later point. They have been warned.

> or at least make it configurable - via yaml?

No, this will not be made configurable. It will be fixed. It should never have been created this way, but it's been like this since the very beginning of this repository.

> however, I don't know whether this is generally wrong or depend on robot type and/or individual settings for each robot...

The orientation of the `base_link` mesh is incorrect for all non-e-Series robot models afaik.

**Comment by fmessmer** [comment]
> > but it's misleading
> 
> "misleading" is a bit strong. It's confusing. It's not like this has been purposefully modelled like this, to make things harder for everyone else.

I meant to say it that way, i.e. "confusing"

> 
> > if you get hints from the support forum as posted above
> 
> I still wouldn't know what you are referring to, as we cannot read the discussion.

I've added them in the top post of this issue

> > > NOTE: this WILL be removed in a future revision of this file
> > 
> > 
> > why not now together with all the other "breaking" changes?
> 
> Because I haven't had time for it yet, together with the many other things that still need to be done.
> 
> And I want to do it in its own PR, not as part of some other PR.

I didn't mean to complain - I can provide a PR, I justed wanted to discuss how it should be done

> The comment is just there as a fail-safe: in case this is forgotten, or people start using `melodic-devel-staging` before they should, no one can complain when it _is_ fixed at some later point. They have been warned.
> 
> > or at least make it configurable - via yaml?
> 
> No, this will not be made configurable. It will be fixed. It should never have been created this way, but it's been like this since the very beginning of this repository.

so you want the meshes (`base.dae` and `base.stl`) untouched but remove the rotation between `base_link` and `base`, correct?
do you then still want to keep both frames for compatibility - as they would then be identical afterwards (keeping would be my preferred option, too)

> > however, I don't know whether this is generally wrong or depend on robot type and/or individual settings for each robot...
> 
> The orientation of the `base_link` mesh is incorrect for all non-e-Series robot models afaik.

**Comment by fmessmer** [comment]
the PR were this got merged was https://github.com/ros-industrial/universal_robot/pull/200

**Comment by gavanderhoorn** [comment]
> the PR were this got merged was #200

That PR introduces the `base` and `tool0` frames, but not the orientation problem with the `base_link` mesh.

The rotation introduced there is exactly to compensate for the incorrect orientation of `base_link` when compared with the model shown on the TP.

**Comment by gavanderhoorn** [comment]
I'm going to assume this was addressed by #505 and close this.

If it wasn't, please comment here and we can re-open.
